login/service.go

```diff
@@ -678,17 +764,33 @@ func (keycloak *KeycloakOAuthProvider) CreateOrUpdateKeycloakUser(accessToken st
 				"username":             claims.Username,
 				"err":                  err,
 			}, "unable to create user/identity")
-			return nil, nil, errors.New("failed to create user/identity " + err.Error())
+			return errors.New("failed to create user/identity " + err.Error())
 		}
 
 	} else {
-		identity = &identities[0]
-		user = &identity.User
+		// identity = &identities[0]  - this wont work because we need pass-by-reference to work
```

Comment:
One more reason to return Identity & User instead of updating parameters.

Reply:
done in https://github.com/fabric8-services/fabric8-auth/pull/82/commits/d596a03b4deaef6b7e560c3fac124db171b6b7c5